friends/src/App.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux'
+import { fetchFriends, addFriend } from './actions'
+import './App.css';
+
+class App extends Component {
+
+  componentDidMount() {
+    this.props.fetchFriends();
+  }
+
+  handleInput = e => {
+    this.setState({ [e.target.name]: e.target.value })
+  }
+
+
+  render() {
+    return (
+      <div className="App">
+      {this.props.friends.map(friends => {
+        return(
+          <div key={friends.id}>{friends.name}, {friends.age}, email: {friends.email}</div>)
+      })}
+
+      <div className='form'>
+        <input 
+        type='text'
+        onChange={this.handleInput}
+        placeholder='name'
+        name='name'
+        />
+        <input 
+        type='number'
+        onChange={this.handleInput}
+        placeholder='age'
+        name='age'
+        />
+        <input 
+        type='text'
+        onChange={this.handleInput}
+        placeholder='email'
+        name='email'
+        />
+        <button onClick={this.props.addFriend(this.props.friends.id)}>Save</button>
```

Comment:
Hmm how would You want to add a friend by just passing an id? This functionality needs some overhaul. First of all i would advise creating `refs` for the input field values. Then create a function that will addFriend and use those ref values to crate a friend object that You can then pass to the function from props and send it to actions.